packages/forklift-console-plugin/src/modules/Plans/views/details/components/MappingListItem.tsx

```diff
@@ -129,13 +134,30 @@ export const MappingListItem: FC<MappingListItemProps> = ({
             aria-label={t('Actions')}
             aria-labelledby=""
           >
-            <Button
-              onClick={onClick}
-              variant="plain"
-              aria-label={t('Delete mapping')}
-              key="delete-action"
-              icon={<MinusCircleIcon />}
-            />
+            {!canEditMapping ? (
+              <Tooltip
+                content={t(
+                  `This mapping's associated VM has been migrated and can no longer be modified.`,
+                )}
+              >
+                <Button
+                  onClick={onClick}
+                  variant="plain"
+                  aria-label={t('Delete mapping')}
+                  key="delete-action"
+                  icon={<MinusCircleIcon />}
+                  isAriaDisabled
+                />
+              </Tooltip>
+            ) : (
+              <Button
+                onClick={onClick}
+                variant="plain"
```

Comment:
u can use ButtonVariant from PF

Reply:
This is pre-existing code? I would say these kinds of changes are generally out-of-scope

Reply:
You are right, but its a process of bringing this project to a better level

Reply:
on second thought, is it necessary? we have typescript types